test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +92,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(isMutating) {
```

Comment:
This change does not help much, as it does not confirm that saving happens *only* for mutating commands. Tests will pass even if you did not change this. To be more complete, you should add a branch to confirm that the file is not updated for non-mutating commands.